arrow-core-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -1,42 +1,114 @@
+      Law("Foldable Laws: reduceLeftToOption combines all items into an optional value") { FF.`reduceLeftToOption returns Option value`(GENB, EQOptionInt) },
+      Law("Foldable Laws: reduceRightToOption combines all items into an optional value") { FF.`reduceRightToOption returns Option value`(GENB, EQOptionInt) },
+      Law("Foldable Laws: reduceLeftOption consistent with reduceLeftToOption") { FF.`reduceLeftOption returns Option value`(GEN, EQOptionInt) },
+      Law("Foldable Laws: reduceRightOption consistent with reduceRightToOption") { FF.`reduceRightOption returns Option value`(GEN, EQOptionInt) },
+      Law("Foldable Laws: traverse_ consistent with foldRight") { FF.`traverse consistent with foldRight`(Id.applicative(), GEN, Id.genK(), Id.eqK().liftEq(Eq.any())) },
+      Law("Foldable Laws: sequence_ consistent with traverse_") { FF.`sequence consistent with traverse`(Id.applicative(), GENK, Id.genK(), Id.eqK().liftEq(Eq.any())) },
+      Law("Foldable Laws: isEmpty returns if there are elements or not") { FF.`isEmpty returns if there are elements or not`(GEN, EQBool) },
+      Law("Foldable Laws: isNotEmpty consistent with isEmpty") { FF.`isNotEmpty consistent with isEmpty`(GEN, EQBool) },
+      Law("Foldable Laws: foldMapM folds on F mapping values to G(B) using given Monoid") { FF.`foldMapM folds on F mapping values to G(B) using given Monoid`(GEN, EQForIdInt) },
+      Law("Foldable Laws: get gets the item at the given index of the Foldable") { FF.`get gets the item at the given index of the Foldable`(GEN, EQOptionInt) }
+    )
+  }
+
+  fun <F> laws(FF: Foldable<F>, GA: Applicative<F>, EQK: EqK<F>): List<Law> {
+    val EQKListKInt: Eq<Kind<F, Kind<ForListK, Int>>> = EQK.liftEq(ListK.eq(Int.eq()))
 
     return listOf(
-        Law("Foldable Laws: Left fold consistent with foldMap") { FF.leftFoldConsistentWithFoldMap(GEN, EQ) },
-        Law("Foldable Laws: Right fold consistent with foldMap") { FF.rightFoldConsistentWithFoldMap(GEN, EQ) },
-        Law("Foldable Laws: Exists is consistent with find") { FF.existsConsistentWithFind(GEN) },
-        Law("Foldable Laws: Exists is lazy") { FF.existsIsLazy(GEN, EQ) },
-        Law("Foldable Laws: ForAll is lazy") { FF.forAllIsLazy(GEN, EQ) },
-        Law("Foldable Laws: ForAll consistent with exists") { FF.forallConsistentWithExists(GEN) },
-        Law("Foldable Laws: ForAll returns true if isEmpty") { FF.forallReturnsTrueIfEmpty(GEN) },
-        Law("Foldable Laws: FirstOption returns None if isEmpty") { FF.firstOptionReturnsNoneIfEmpty(GEN) },
-        Law("Foldable Laws: FirstOption returns None if predicate fails") { FF.firstOptionReturnsNoneIfPredicateFails(GEN) },
-        Law("Foldable Laws: FoldM for Id is equivalent to fold left") { FF.foldMIdIsFoldL(GEN, EQ) }
-      )
+      Law("Foldable Laws: orEmpty consistent with just empty") { FF.`orEmpty consistent with just empty`(GA, EQKListKInt) }
+    )
```

Comment:
This never tests any of the other laws right? Or am I missing something here?

Reply:
Mmm aren't the other laws tested through a different path? or I should call them from here too??

Reply:
Well if they are fine, but please test that because it does not look like it to me. Iirc in most other tests where we have additional tests for additional instances we call them in this place.